xarray/core/indexing.py

```diff
@@ -1291,9 +1302,29 @@ def __getitem__(self, indexer):
 
         return result
 
-    def transpose(self, order):
+    def transpose(self, order) -> pd.Index:
         return self.array  # self.array should be always one-dimensional
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return ('%s(array=%r, dtype=%r)'
                 % (type(self).__name__, self.array, self.dtype))
+
+    def copy(self, deep: bool = True) -> 'PandasIndexAdapter':
+        obj = object.__new__(PandasIndexAdapter)
```

Comment:
Is it possible to use the constructor directly here rather than `object.__new__`?

If not, please add a comment explaining why :)

Reply:
I can use ``__init__``, but it's going to be needlessly slower. If you prefer I can add a fastpath=False parameter to ``__init__`, which is the pattern already used elsewhere

Reply:
I can use ``__init__``, but it's going to be needlessly slower. If you prefer I can add a fastpath=False parameter to ``__init__`, which is the pattern already used elsewhere

Reply:
I would rather stick with init unless we can find an example of using xarray's public APIs where performance is meaningfully effected. Until then it's probably premature optimization.

Reply:
Done